raspberry/read_sbus/radioDataParser.py

```diff
@@ -0,0 +1,122 @@
+RADIO_TRIMMER_MAX_OMEGA_Z_GAIN = 0.05
+
+
+RADIO_CHANNEL_YAW_RATE = 0
+RADIO_CHANNEL_PITCH = 1
+RADIO_CHANNEL_THROTTLE = 2
+RADIO_CHANNEL_ROLL = 3
+RADIO_CHANNEL_VRA = 5
+RADIO_CHANNEL_VRB = 7
+RADIO_CHANNEL_VRC = 6
+RADIO_CHANNEL_VRE = 11
+RADIO_CHANNEL_SWITCH_A = 9
+RADIO_CHANNEL_SWITCH_B = 8
+RADIO_CHANNEL_SWITCH_C = 4
+RADIO_CHANNEL_SWITCH_D = 10
+
+DOWN = "DOWN"
+MID = "MID"
+UP = "UP"
+
+
+
+ZERO_ROTOR_SPEED = 1000
+ABSOLUTE_MAX_PWM = 1900
+
+
+class RadioDataParser:
+    def __init__(self):
+        self.m_channelData = None
+        self.rawRatePercentage = None
+        self.switchValue = None
+        self.ThreeWaySwitch = {
+            "DOWN" : 0,
+            "MID" : 1,
+            "UP" : 2
+        }
+
+    def mapRadioToAngle(self, x):
+        return -PITCH_MAX_RAD + (float)(x - RADIO_STICK_MIN) / ((float)(RADIO_STICK_MAX - RADIO_STICK_MIN)) * 2 * PITCH_MAX_RAD
+
+    def mapTrimmerToPercentage(self, x):
+        percentage = (float)(x - RADIO_STICK_MIN) / ((float)(RADIO_STICK_MAX - RADIO_STICK_MIN))
+        percentage = 1 if percentage > 1 else percentage
+        return percentage if percentage >= 0 else 0
+
+    def pitchReferenceAngleRad(self):
+        return self.mapRadioToAngle(self.m_channelData[RADIO_CHANNEL_PITCH])
+
+    def rollReferenceAngleRad(self):
+        return self.mapRadioToAngle(self.m_channelData[RADIO_CHANNEL_ROLL])
+
+    def yawRateReferenceRadSec(self):
+        self.rawRatePercentage = self.mapTrimmerToPercentage(self.m_channelData[RADIO_CHANNEL_YAW_RATE]);
+        return -RADIO_MAX_YAW_RATE_RAD_SEC + RADIO_MAX_YAW_RATE_RAD_SEC * self.rawRatePercentage
+    
+    def throttleReferencePercentage(self):
+        return self.mapTrimmerToPercentage(self.m_channelData[RADIO_CHANNEL_THROTTLE])
+
+    def armed(self):
+        return self.m_channelData[RADIO_CHANNEL_SWITCH_B] >= 1500
+
+    def kill(self):
+        return self.m_channelData[RADIO_CHANNEL_SWITCH_A] >= 1500
+
+    def switchC(self):
+        self.switchValue = self.m_channelData[RADIO_CHANNEL_SWITCH_C];
+        if self.switchValue <= 450:
+            return self.ThreeWaySwitch[DOWN]
+        elif self.switchValue <= 1200:
+            return self.ThreeWaySwitch[MID]
+        return self.ThreeWaySwitch[UP]
+
+    def switchD(self):
+        return self.m_channelData[RADIO_CHANNEL_SWITCH_D] >= 1500
+
+    def trimmerVRAPercentage(self):
+        return self.mapTrimmerToPercentage(self.m_channelData[RADIO_CHANNEL_VRA])
+
+    def trimmerVRCPercentage(self):
+        return self.mapTrimmerToPercentage(self.m_channelData[RADIO_CHANNEL_VRC])
+
+    def trimmerVRBPercentage(self):
+        return self.mapTrimmerToPercentage(self.m_channelData[RADIO_CHANNEL_VRB])
+
+    def trimmerVREPercentage(self):
+        return self.mapTrimmerToPercentage(self.m_channelData[RADIO_CHANNEL_VRE])
+    
+    def mapPrcnt(self, percentage, minVal, maxVal):
+        return minVal + percentage * (maxVal - minVal)
+    
+    def encapsulateRadioData(self):
```

Comment:
Can you do something to improve the readability of `encapsulateRadioData`?

Reply:
Let me know if this is okay and if I can resolve this comment.